core/execution/engine.go

```diff
@@ -250,6 +250,9 @@ func (e *Engine) RejectMarket(ctx context.Context, marketID string) error {
 		return err
 	}
 
+	// send market data event so market data and markets API are consistent.
+	e.broker.Send(events.NewMarketDataEvent(ctx, mkt.GetMarketData()))
```

Comment:
Why aren't we sending this in `mkt.Reject(ctx)` so that its alongside the other event thats sent? Doesn't make sense for it to be here in a different place.

Reply:
This event is sent because rejected successor markets get skipped in the `OnTick` calls, where all other markets send a market data event. Because that event isn't being sent, the data-node API's are a bit borked. This is sending the event for rejected markets, to complete the market data events from the OnTick loop in execution engine, so I figured it makes sense to keep it in execution engine. The market updated event being sent in the `mkt.Reject()` call is arguably messy, because the execution engine is where we send a bunch of these events (aka `publishNewMarketInfos()` etc...), but realistically, it'll always be a spread across both, to account for oracles (trading terminated and settled)